Data-Structure-Arrays.md

```diff
@@ -4,6 +4,49 @@ Internally, `array` is a kind of data structure that can store a fixed-size sequ
 
 `array` consists of contiguous memory locations. The lowest address corresponds to the first element and the highest address to the last element.
 
+
+## Arrays in C++
+
+C++ provides a data structure, `array`, which stores a fixed-size sequential collection of elements of the same data-type. An `array` is used to store a collection of data, but it is better to think of an `array` as a collection of variables of the same type.
+
+#### Declaration of `array`
+
+```c++
+int intarray[10]; // Declares an array of integer type of size 10 with elements having random values. Index ranges from 0 to 9(i.e. size-1).
+int intarray[10] = { 0 }; // Declares an array of integer of size 10 with all elements having value 0
+
+// Choose one the two declarations and then move ahead.
+```
+
+#### Inserting elements to `array`:
+
+```c++
+intarray[0] = 1; // Inserts an integer value of 1 at index 0
+intarray[1] = 0; // Inserts an integer value of 0 at index 1
+intarray[2] = -1; // Inserts an integer value of -1 at index 2
+intarray[3] = 1; // Inserts an integer value of 1 at index 3
+```
+
+#### Printing an `array`:
+
+```c++
+std::cout << intarray[0] << std::endl; // Returns 1 which is element at index of the array
+std::cout << intarray[11] << std::endl; // A random number is expected, while in reality this is `dangerous`, and is primary cause of crashes as it's accessing a memory location which does not exist.
```

Comment:
I think you mis-interpreted [this comment](https://github.com/FreeCodeCamp/wiki/pull/1056#discussion_r65341995).

This should be:
```diff
-std::cout << intarray[11] << std::endl; // A random number is expected, while in reality this is `dangerous`, and is primary cause of crashes as it's accessing a memory location which does not exist.
+std::cout << intarray[11] << std::endl; // Would give a a "Garbage" value as there is no element at index 11 of array. That memory location is beyond the range of the array.

```